k8s-model-server/README.md

```diff
@@ -0,0 +1,170 @@
+example demonstrates how to serve a model from distributed storage.  [GKE](https://cloud.google.com/kubernetes-engine/) 
+(Google Kubernetes Engine) and [GCS](https://cloud.google.com/storage/) (Google Cloud Storage) are used here, 
+but the example is trivially adaptable to an on-premises (on-prem) cluster using NFS (Network File System) or 
+HDFS (Hadoop Distributed File System), or an AWS (Amazon Web Services)cluster using S3 (Simple Storage Service).
+
+## Create Google Cloud Platform (GCP) Project 
+This tutorial uses [GCP](https://cloud.google.com/).
+[This documentation](https://cloud.google.com/resource-manager/docs/creating-managing-projects) demonstrates how to 
+create and manage a new Project on GCP. There is a [free tier](https://cloud.google.com/free/) available that you can 
+use to try it out.
+
+## Build a Model Server Docker Image
+The [docker file](docker/Dockerfile) builds a docker image containing default the TensorFlow model server tuned for 
+amd64 architecture. You can modify this file to support the architecture of your choice.
+
+### Set the User (Optional)
+Note the line at the beginning of the Dockerfile.
+
+```
+ENV MS_USER=model-server
+```
+
+This is the user that the model-server process will run as. You can modify this user to change the name. This user 
+will be used in the Kubernetes Deployment's PodSecurityContext so that the container is not run with root privileges
+(This is considered to be a best practice).
+
+### Build the Image
+A [Makefile](docker/Makefile) is included that will build the image and push it to private hosting inside your project. 
+
+Download and install the [Google Cloud SDK](https://cloud.google.com/sdk/downloads) (Software Development Kit).
+
+Enable the [GCR](https://cloud.google.com/container-registry/docs/quickstart) (Google Container Registry) API in 
+your project.
+
+Set the name of your project at the top of the Makefile ```PROJECT_ID=<your-project>```.
+
+The command below will build the docker image.
+
+```commandline
+make all
+```
+
+The command below will push the docker image to your project's GCR.
+
+```commandline
+make push
+```
+
+You can learn more about [building a TensorFlow model server](https://www.tensorflow.org/serving/serving_advanced) and 
+running a [TensorFlow model server in docker](https://www.tensorflow.org/serving/docker) in the TensorFlow Serving 
+documentation.
+
+## Create a Bucket
+You are going to serve the model from [GCS](https://cloud.google.com/storage/). 
+
+Download and install the [Google Cloud SDK](https://cloud.google.com/sdk/downloads) (Software Development Kit).
+
+Use the [gsutil mb](https://cloud.google.com/storage/docs/gsutil/commands/mb) command to create a bucket. Note that 
+the bucket name must be globally unique.
+```commandline
+gsutil mb gs://<ucket-name>
+``` 
+
+
+## Upload a Model
+You can train an inception model using the instructions from the 
```

Comment:
Mention that inception is an example, but the workflow is adaptable to any TF model. It currently reads as though the guide is only for inception.

Reply:
+1